lms/djangoapps/bulk_email/tasks.py

```diff
@@ -400,14 +402,17 @@ def _get_source_address(course_id, course_title):
     """
     course_title_no_quotes = re.sub(r'"', '', course_title)
 
-    # For the email address, get the course.  Then make sure that it can be used
-    # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
-    # character appears.
-    from_addr = u'"{0}" Course Staff <{1}-{2}>'.format(
-        course_title_no_quotes,
-        re.sub(r"[^\w.-]", '_', course_id.course),
-        settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
+    # Translators: Bulk email from address e.g. ("Physics 101" Course Staff)
+    course_staff_title = _('"{course_name}" Course Staff')
+
+    from_addr_format = course_staff_title + u' <{course_id}-{from_email}>'
```

Comment:
Well, GitHub didn't help because it almost hardcodes the `direction: ltr`. I've created a [fiddle](https://plnkr.co/edit/QbXj4jqj3V32EFgatBkJ?p=preview) to show that how the email shows properly for the Arabic user although the byte-order is the same.

I still thinks that the concatenation is *the correct* method shouldn't have another way.